src/main/java/com/blackducksoftware/tools/commonframework/core/config/ConfigurationProperties.java

```diff
@@ -119,6 +125,11 @@ public void addAll(final Properties sourceProps) {
 
 		for (final Object keyObj : sourceProps.keySet()) {
 			final String key = (String) keyObj;
+			if (key.equals("cc.password")){
+				logger.debug("addAll(): adding: " + key + "=" + "******");
+				config.addProperty(key, sourceProps.getProperty(key));
+				continue;
+			}
 			logger.debug("addAll(): adding: " + key + "=" + sourceProps.getProperty(key));
 			config.addProperty(key, sourceProps.getProperty(key));
 		}
```

Comment:
I have basically the observations as above, but since it's happening twice, maybe it's better to have a new method:

private String obscurePassword(key, value) {
  if (key.endsWith(".password") {
    return "********";
  return value;
}

Then both logger stmts could look like:
logger.debug(.. key + "=" + obscurePassword(key, obscurePassword(sourceProps.getProperty(key)));
and the only new conditional is the one in obscurePassword()